src/Service.Tests/SqlTests/GraphQLSupportedTypesTests/MySqlGQLSupportedTypesTests.cs

```diff
@@ -24,27 +26,135 @@ public static async Task SetupAsync(TestContext context)
+        [DataRow("1000-01-01", "1000-01-01T00:00:00.000Z", DisplayName = "Min date for datetime column stored with zeroed out time.")]
+        [DataTestMethod]
+        public async Task InsertMutationInput_DateTimeTypes_ValidRange_ReturnsExpectedValues(string dateTimeGraphQLInput, string expectedResult)
+        {
+            // Arrange
+            const string DATETIME_FIELD = "datetime_types";
+            string graphQLMutationName = "createSupportedType";
+            string gqlMutation = "mutation{ createSupportedType (item: {" + DATETIME_FIELD + ": \"" + dateTimeGraphQLInput + "\" }){ typeid, " + DATETIME_FIELD + " } }";
+
+            // Act
+            JsonElement actual = await ExecuteGraphQLRequestAsync(gqlMutation, graphQLMutationName, isAuthenticated: true);
+
+            // Assert
+            Assert.IsTrue(
+                condition: actual.GetProperty("typeid").TryGetInt32(out _),
+                message: "Error: GraphQL mutation result indicates issue during record creation because primary key 'typeid' was not resolved.");
+
+            Assert.AreEqual(
+                expected: expectedResult,
+                actual: actual.GetProperty(DATETIME_FIELD).GetString(),
+                message: "Unexpected datetime value.");
+        }
+
+        /// <summary>
+        /// For MySQL, values passed to columns with datatype datetime(0) (no fractional seconds) that only include time
+        /// are auto-populated with the current date.
+        /// Based on the supplied time (assumed to be UTC), gets the current date. This calculation must fetch
+        /// DateTime.UtcNow, otherwise the test machine's timezone will be used and result in an unexpected date.
+        /// </summary>
+        /// <param name="dateTimeGraphQLInput">Unescaped string used as value for GraphQL input field datetime_types</param>
+        /// <param name="expectedResult">Expected result the HotChocolate returns from resolving database response.</param>
+        [DataRow("23:59:59.499999", "23:59:59.000Z", DisplayName = "hh:mm::ss.ffffff for datetime column stored with zeroed out date and rounded down fractional seconds.")]
+        [DataRow("23:59:59", "23:59:59.000Z", DisplayName = "hh:mm:ss for datetime column stored with zeroed out date.")]
+        [DataRow("23:59", "23:59:00.000Z", DisplayName = "hh:mm for datetime column stored with zeroed out date and seconds.")]
+        [DataTestMethod]
+        public async Task InsertMutationInput_DateTimeTypes_TimeOnly_ValidRange_ReturnsExpectedValues(string dateTimeGraphQLInput, string expectedResult)
+        {
+            // Arrange
+            expectedResult = DateTime.UtcNow.ToString("yyyy-MM-ddT") + expectedResult;
+            await InsertMutationInput_DateTimeTypes_ValidRange_ReturnsExpectedValues(dateTimeGraphQLInput, expectedResult);
+        }
+
+        /// <summary>
+        /// MySql Single Type Tests.
+        /// </summary>
+        /// <param name="type">GraphQL Type</param>
```

Comment:
maybe GraphQL DataType, to not get confused with GraphQL request type:query/mutation

Reply:
updated